python/paddle/distributed/launch.py

```diff
@@ -117,33 +71,128 @@ def parse_args():
 PADDLE_TRAINER_ENDPOINTS
 POD_IP (current node ip address, not needed for local training)
 ''')
+
+    # Optional arguments for the launch helper
     parser.add_argument(
-        '--gpus',
+        "--node_ips",
+        type=str,
+        default="127.0.0.1",
+        help="Paddle trainer ips, such as 192.168.0.16,192.168.0.17..")
+
+    parser.add_argument(
+        "--node_id",
         type=int,
-        default=8,
-        help='start number of processes for every gpu')
+        default=0,
+        help="The trainer id of the node for multi-node distributed "
+        "training")
+
+    parser.add_argument(
+        "--print_config",
+        type=bool,
+        default=True,
+        help="Print the config or not")
+
     parser.add_argument(
-        '--log_dir',
+        "--selected_gpus",
         type=str,
-        default="mylog",
-        help='directory to put logs per process.')
+        default="0,1,2,3,4,5,6,7",
+        help="It's for gpu trainning and the trainning process will run on the selected_gpus,"
+        "each process is bound to a single GPU.")
+
+    parser.add_argument(
+        "--split_log_path", type=str, help="The splited log path.")
+
+    # positional
     parser.add_argument(
-        'entrypoint_script',
+        "training_script",
         type=str,
-        help="The entrypoint script to be launched in parallel,"
-        "followed by all the arguments for each process,"
-        "e.g. train.py --lr 0.1")
-    parser.add_argument('entrypoint_args', nargs=argparse.REMAINDER)
+        help="The full path to the single GPU training "
+        "program/script to be launched in parallel, "
+        "followed by all the arguments for the "
+        "training script")
+
+    # rest from the training program
+    parser.add_argument('training_script_args', nargs=REMAINDER)
     return parser.parse_args()
 
 
-def main():
-    args = parse_args()
+def start_procs(args):
+    """
+    """
+    procs = []
+    log_fns = []
+
+    default_env = os.environ.copy()
+
+    node_id = args.node_id
+    node_ips = [x.strip() for x in args.node_ips.split(',')]
+    assert node_id >= 0 and node_id < len(node_ips), \
+            "node_id:{} must be in range of the node_ips:{}".format(node_id, node_ips)
```

Comment:
"node_id:{} must be in range of the SIZE OF node_ips:{}".format(node_id, LEN(node_ips))

Reply:
Done.